src/marks/halo.js

```diff
@@ -0,0 +1,33 @@
+import {isColor} from "../options.js";
+
+let nextHaloId = 0;
+
+export function applyHalo(g, {color, radius}) {
+  const id = `plot-linehalo-${nextHaloId++}`;
+  g.selectChildren().style("filter", `url(#${id})`);
+  g.append("filter").attr("id", id).html(`
+    <feMorphology in="SourceAlpha" result="DILATED" operator="dilate" radius="${radius}"></feMorphology>
+    <feFlood flood-color="${color}" result="BG"></feFlood>
+    <feComposite in="BG" in2="DILATED" operator="in" result="OUTLINE"></feComposite>
+    <feMerge>
+      <feMergeNode in="OUTLINE" />
+      <feMergeNode in="SourceGraphic" />
+    </feMerge>`);
```

Comment:
You only need the result+in for *dilated*; otherwise in defaults to the previous filter primitive.

```suggestion
    <feMorphology in="SourceAlpha" result="dilated" operator="dilate" radius="${radius}"></feMorphology>
    <feFlood flood-color="${color}"></feFlood>
    <feComposite in2="dilated" operator="in"></feComposite>
    <feMerge>
      <feMergeNode></feMergeNode>
      <feMergeNode in="SourceGraphic"></feMergeNode>
    </feMerge>`);
```